### PR TITLE
[Refactor] cleanup lv utils

### PR DESCRIPTION
**Pull request by Tastyep**
# Description

#### Motivation:
I think lv-utils is a big amalgam of code that we don't know where to put and is becoming uglier the bigger it grows.
Additionally, there is code interpreted at load time which makes `require "lv-utils"` from init.lua hard to understand (what does it do and why do I have to switch files to understand).
Finally, since we renamed `lv-plugins` I thought it coherent to rename `lv-utils` to `utils` (do we agree that this lv- prefix should not be there?).

#### What it does:
I always advise to check commit by commit for that :)

### CODE REVIEW

File: lua/augroups.lua [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  -- _gemini = {
+  --     {'BufWinEnter', '.gmi', 'setlocal filetype=markdown'}, {'BufRead', '*.gmi', 'setlocal filetype=markdown'},
+  --     {'BufNewFile', '*.gmi', 'setlocal filetype=markdown'}
+  -- },
+  _markdown = {
+    { "FileType", "markdown", "setlocal wrap" },
+    { "FileType", "markdown", "setlocal spell" },
+  },
+  _buffer_bindings = {
+    { "FileType", "floaterm", "nnoremap <silent> <buffer> q :q<CR>" },
```
Comment (abzcoding) [inline lua/augroups.lua:49]: while at it, can we remove this :)
there is no float term anymore

File: lua/keymappings.lua [hunk 2]
```diff
@@ -103,11 +103,11 @@ local function get_user_keys(mode)
...
-lv_utils.add_keymap_normal_mode(opts.nnoremap, get_user_keys "normal_mode")
-lv_utils.add_keymap_insert_mode(opts.inoremap, get_user_keys "insert_mode")
-lv_utils.add_keymap_visual_mode(opts.vnoremap, get_user_keys "visual_mode")
-lv_utils.add_keymap_visual_block_mode(opts.xnoremap, get_user_keys "visual_block_mode")
-lv_utils.add_keymap_term_mode(opts.generic, get_user_keys "visual_block_mode")
+utils.add_keymap_normal_mode(opts.nnoremap, get_user_keys "normal_mode")
+utils.add_keymap_insert_mode(opts.inoremap, get_user_keys "insert_mode")
+utils.add_keymap_visual_mode(opts.vnoremap, get_user_keys "visual_mode")
+utils.add_keymap_visual_block_mode(opts.xnoremap, get_user_keys "visual_block_mode")
+utils.add_keymap_term_mode(opts.generic, get_user_keys "visual_block_mode")
```
Comment (abzcoding) [inline lua/keymappings.lua:110]: also this needs to be `term_mode` instead of `visual_block_mode`

File: lua/augroups.lua [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+  },
+  _auto_resize = {
+    -- will cause split windows to be resized evenly if main window is resized
+    { "VimResized", "*", "wincmd =" },
+  },
+  _packer_compile = {
+    -- will cause split windows to be resized evenly if main window is resized
```
Comment (abzcoding) [inline lua/augroups.lua:56]: i know this is unrelated, and i'm sorry for asking this, but can you fix this comment as well :pleading_face:

**Comment by abzcoding** [comment]
also, can you update the `README.md` as well inside your own branch, cause there are "lv-utils" in lines 84 and 159

**Comment by Tastyep** [comment]
Will redo later

**Pull request closed without merging**